openedx/openedx-platform
# BB-986: Add variable that allows overriding marketing urls

This PR adds a ways to enable overriding any marketing URL, either by configuration on `lms.env.json` or `SiteConfiguration`.

**JIRA tickets**: [OSPR-3340](https://openedx.atlassian.net/browse/OSPR-3340)

**Sandbox URL**: https://pr20378.sandbox.opencraft.hosting/

**Merge deadline**: None.

**Testing instructions**:
1. Pull this branch on you local devstack.
2. Run tests on LMS shell with `pytest common/djangoapps/edxmako/tests.py::ShortcutsTests`
3. Add `MKTG_URL_OVERRIDES` on `lms.env.json`:
```
....
"MKTG_URL_OVERRIDES": {
    "TOS": "https://www.edx.org"
},
...
```
4. Restart LMS and memcached (`make lms-restart && docker-compose restart memcached`) and check if the Terms of service link redirects to www.edx.org.
5. Change the override to a invalid URL (like `12345`) and check that the link disappears.
6. Redo steps 3..5 but setting `MKTG_URL_OVERRIDES` on `SiteConfiguration` instead of in `lms.env.json`.
7. Check that the TOS link on the sandbox leads to https://www.edx.org/edx-terms-service :smile: 

**Author notes and concerns**:

1. If not set, this doesn't alter the default behaviour, but setting it makes it have precedence over any configuration.

**Reviewers**

**Settings**
```yaml
RETIREMENT_SERVICE_SETUP: false

EDXAPP_LMS_ENV_EXTRA:
  MKTG_URL_OVERRIDES:
    PRIVACY: https://www.edx.org/edx-terms-service
```
